### PR TITLE
Bump got from 11.8.5 to 11.8.6

### LINKED ISSUE #445: [BUG] oncancel handler error on macos-latest
**Describe the bug**
i'm getting an error running the action on a macos-latest image, i tried with the v2 action and even the main version,  having the same issue

**To Reproduce**
```yaml
    - name: Import Workflow Secrets
      uses: hashicorp/vault-action@main
      with:
        url: ${{ inputs.vaultUrl }}
        method: approle
        roleId: ${{ inputs.roleId }}
        secretId: ${{ inputs.secretId }}
        secrets: ${{ inputs.secrets }}
        exportToken: true
```

**Expected behavior**
it should just work :)

**Log Output**
For the most verbose logs, [add a secret called `ACTIONS_STEP_DEBUG` with the value `true`](https://github.com/actions/toolkit/blob/main/docs/action-debugging.md). Then, re-run the workflow if possible and post the *raw logs* for the step here with any sensitive information masked or removed.
```
/Users/runner/work/_actions/hashicorp/vault-action/main/dist/index.js:17305
					throw new Error('The `onCancel` handler was attached after the promise settled.');
					      ^

Error: The `onCancel` handler was attached after the promise settled.
    at onCancel (/Users/runner/work/_actions/hashicorp/vault-action/main/dist/index.js:17305:12)
    at makeRequest (/Users/runner/work/_actions/hashicorp/vault-action/main/dist/index.js:3622:13)
    at Request.<anonymous> (/Users/runner/work/_actions/hashicorp/vault-action/main/dist/index.js:3727:17)
    at Object.onceWrapper (node:events:642:26)
    at Request.emit (node:events:527:28)
    at Timeout.retry (/Users/runner/work/_actions/hashicorp/vault-action/main/dist/index.js:5251:30)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
```

**Additional context**
it seems to work only once, then addtional attempts just fail :(

### PR DESCRIPTION
**Pull request by nafarlee**
Closes #445

The cause was an underlying issue in `got` (https://github.com/sindresorhus/got/issues/1489) which was addressed by a PR back in November (https://github.com/sindresorhus/got/pull/2187). [`got@v11.8.6`](https://github.com/sindresorhus/got/releases/tag/v11.8.6) contains the fix and is a simple upgrade, especially when compared to upgrading to [`got@12.x`](https://github.com/sindresorhus/got/releases/tag/v12.0.0) or [`got@13.x`](https://github.com/sindresorhus/got/releases/tag/v13.0.0) (as recommended by #457).

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -3702,6 +3702,7 @@ function asPromise(normalizedOptions) {
                     request._beforeError(new types_1.HTTPError(response));
                     return;
                 }
+                request.destroy();
```
Comment (fairclothjm) [inline dist/index.js:3705]: Please revert the changes to `dist/index.js`. We do not commit directly to this file except when the team is doing a release.

Comment (nafarlee) [reply dist/index.js]: Ah, my apologies, I didn't realize! I pushed a commit removing this particular change

**Comment by nafarlee** [comment]
@maxcoulombe Glad I could help!

**Comment by ReinAchten-TomTom** [comment]
> Hey @nafarlee thanks for taking care of this! The fix will be included in the upcoming release which will hopefully address the issue for good for everyone. Cheers.

Hi @maxcoulombe (apologies for "reviving" a closed pull request), when is the next upcoming release planned? We're bit by the problem solved by this change regularly, and it'd be great if we could benefit from this change. Thnx!
